extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/OpenTelemetryVertxMetricsFactory.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.opentelemetry.runtime.tracing.vertx;
+
+import io.vertx.core.VertxOptions;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.core.spi.VertxMetricsFactory;
+import io.vertx.core.spi.metrics.HttpServerMetrics;
+import io.vertx.core.spi.metrics.VertxMetrics;
+
+public class OpenTelemetryVertxMetricsFactory implements VertxMetricsFactory {
```

Comment:
Same here. Just to make sure relationship (or presence/absence) of this support w.r.t. micrometer extension is clear

Reply:
I might also consider making OpenTelemetryHttpServerMetrics an inner class so that it is a single unit that fills the gap when micrometer isn't around.

Reply:
Sure.